### PR TITLE
v4.0.0

### PR DESCRIPTION
**Pull request by whymarrh**
Refs #46
~Depends on #58, #62, #63~

This PR represents the next major version, v4.

### CODE REVIEW

**Comment by Gudahtt** [comment]
In testing this on the extension, I noticed that the rule `node/no-unsupported-features/es-syntax` was showing errors for ES6 syntax. This seems contrary to what we want from this config :thinking:

**Comment by Gudahtt** [comment]
Also, `node/file-extension-in-import` is currently requiring file extensions for import statements. Is this the default we want? We currently require the opposite - no file extensions.

**Comment by whymarrh** [comment]
Both good finds, thanks @Gudahtt!

**Comment by whymarrh** [comment]
I'll fix those!

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,12 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ## [Unreleased]
 
+## [4.0.0] - 2020-08-24
```
Comment (Gudahtt) [inline CHANGELOG.md:9]: This should be linked at the bottom of the changelog as well

**Review by Gudahtt — CHANGES_REQUESTED** [review]
Probably worth mentioning #68 and #65 as well

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Comment by whymarrh** [comment]
> A new version of the package `@metamask/eslint-config` (4.0.0) was published at 2020-10-20T19:01:32.862Z from
[redacted]. The shasum of this package was `0c1461a49e6abc924baf23063d58050467d235dd`.

**Pull request merged**